modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustAxumServerCodegen.java

```diff
@@ -532,91 +531,125 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
             original = ModelUtils.getReferencedApiResponse(openAPI, original);
 
             // Create a unique responseID for this response, if one is not already specified with the "x-response-id" extension
+            // The x-response-id may have an appended suffix when multiple content types are present.
             if (!rsp.vendorExtensions.containsKey("x-response-id")) {
                 String[] words = rsp.message.split("[^A-Za-z ]");
-
-                // build responseId from both status code and description
                 String responseId = "Status" + rsp.code + (
                         ((words.length != 0) && (!words[0].trim().isEmpty())) ?
                                 "_" + camelize(words[0].replace(" ", "_")) : ""
                 );
                 rsp.vendorExtensions.put("x-response-id", responseId);
             }
 
-            if (rsp.dataType != null) {
-                // Get the mimetype which is produced by this response. Note
-                // that although in general responses produces a set of
-                // different mimetypes currently we only support 1 per
-                // response.
-                String firstProduces = null;
+            List<String> producesTypes = new ArrayList<>();
 
-                if (original.getContent() != null) {
-                    firstProduces = original.getContent().keySet().stream().findFirst().orElse(null);
-                }
+            if (original.getContent() != null) {
+                producesTypes.addAll(original.getContent().keySet());
+            }
 
-                // The output mime type. This allows us to do sensible fallback
-                // to JSON rather than using only the default operation
-                // mimetype.
-                String outputMime;
-
-                if (firstProduces == null) {
-                    if (producesFormUrlEncoded) {
-                        outputMime = formUrlEncodedMimeType;
-                    } else if (producesPlainText) {
-                        if (bytesType.equals(rsp.dataType)) {
-                            outputMime = octetMimeType;
-                        } else {
-                            outputMime = plainTextMimeType;
-                        }
-                    } else {
-                        outputMime = jsonMimeType;
-                    }
-                } else {
-                    if (isMimetypeWwwFormUrlEncoded(firstProduces)) {
-                        producesFormUrlEncoded = true;
-                        producesPlainText = false;
-                    } else if (isMimetypePlain(firstProduces)) {
-                        producesFormUrlEncoded = false;
-                        producesPlainText = true;
-                    } else {
-                        producesFormUrlEncoded = false;
-                        producesPlainText = false;
-                    }
```

Comment:
@jacob-mink-1996 
I think the PR is too big to review. How about support streaming responses only? Support multiple response type in another PR?

Reply:
Hi @jacob-mink-1996 
I’m also hoping to find some time this week or next to work on this feature. l will be pushing commits here and we can work together.